components/omega/doc/design/MeshClass.md

```diff
@@ -0,0 +1,73 @@
+(omega-design-mesh)=
+# Mesh 
+
+## 1 Overview
+
+The mesh class will contain the YAKL arrays which describe the mesh and are used in the computation of the tendency terms in the discrete governing equations.
+
+## 2 Requirements
+
+### 2.1 Requirement: OMEGA will use the previously estabilished MPAS Mesh Spec
+
+The OMEGA mesh information should be compatiable with the [MPAS Mesh Specification](https://mpas-dev.github.io/files/documents/MPAS-MeshSpec.pdf).
+
+### 2.2 Requirement: Functionality is needed to read the mesh on the host and transfer relevant data to the device for comutation
+
+Not all mesh information is required in computing the tendency terms on the device, e.g. lonCell, latCell, etc. 
+However, other arrays will need to allocated and copied to the device for use in tendency computation.
+One option is to template the mesh class so a mesh object can be created for both the host and device, with a method to perform the copy between the two object types.
```

Comment:
This should eventually be moved to the algorithm or design section.

Reply:
I was only referring to the last line starting with "One option..."

Reply:
I'm removing that line because I believe we've ruled out the template approach in the discussions around this. The Decomp class explicitly creates YAKL arrays for both the host and device for a given variable, so I think it makes sense to use that approach here as well.